### PR TITLE
Remove dependencies

### LINKED ISSUE #1: Remove dependencies?
Hi, that looks like a great example for making an `extendr`-powered package, thank you! 

I don't have a direct usecase for this but I suppose several other packages could use it under the hood. `base64enc` doesn't have any deps so I think this package could be more attractive by also having 0 dependencies, is this something you would consider?

From what I can see `rlang` and `cli` should be easy to remove, but `blob` brings several deps with it. Is `blob` really necessary?

### PR DESCRIPTION
**Pull request by etiennebacher**
Close #1. Ready for review @JosiahParry ;)

### CODE REVIEW

**Comment by JosiahParry** [comment]
Currently running into a bug that I'm going to think through how we can address it. `!b.inherits("raw")` does work in our care right now. 

First, the check for the raw class isn't working and two, we're returning an `Rstr::na()` (don't get me started on the oddities of character vectors and the C api) which, instead, we should be returning a `NULL`

```r
library(b64)
x <- encode(c("hello", "world", NA))

y <- lapply(x, charToRaw)

decode(y)
#> <blob[3]>
#> [[1]]
#> <CHARSXP: NA>
#> 
#> [[2]]
#> <CHARSXP: NA>
#> 
#> [[3]]
#> <CHARSXP: NA>
```

**Comment by JosiahParry** [comment]
We can replace the body of the closure with a `try_from()` coupled with a match statement here.

```rust
  let raw = Raw::try_from(b);
  match raw {
      Ok(r) => {
          let decoded = eng.decode(r.as_slice());
          match decoded {
              Ok(d) => Raw::from_bytes(&d).into_robj(),
              Err(_) => ().into_robj(),
          }
      },
      Err(_) => ().into_robj(),
  }
```

``` r
library(b64)
x <- encode(c("hello", "world", NA))

y <- lapply(x, charToRaw)

decode(y)
#> <blob[3]>
#> [[1]]
#> [1] 68 65 6c 6c 6f
#> 
#> [[2]]
#> [1] 77 6f 72 6c 64
#> 
#> [[3]]
#> NULL
```

File: src/rust/src/lib.rs [hunk 1]
```diff
@@ -195,12 +191,11 @@ fn decode_vectorized_(what: Either<Strings, List>, engine: Robj) -> Robj {
...
-            }
             r.into_iter()
                 .map(|(_, b)| {
-                    if b.is_null() {
+                    if !b.inherits("raw") {
+                        Rstr::na().into_robj()
+                    } else if b.is_null() {
                         ().into_robj()
                     } else {
                         let raw: Raw = b.try_into().unwrap();
```
Comment (JosiahParry) [inline src/rust/src/lib.rs]: See related https://github.com/extendr/extendr/issues/687.

Here we should use try_from() and match as it will catch when it is not a `Raw` and I suppose it is also more "idiomatic" rust as well.

```rust
   let raw = Raw::try_from(b);
                    match raw {
                        Ok(r) => {
                            let decoded = eng.decode(r.as_slice());
                            match decoded {
                                Ok(d) => Raw::from_bytes(&d).into_robj(),
                                Err(_) => ().into_robj(),
                            }
                        },
                        Err(_) => ().into_robj(),
                    }
                })
```

File: README.md [hunk 2]
```diff
@@ -54,28 +57,50 @@ Both `encode()` and `decode()` are vectorized.
...
-#> [5] "QWRpcGlzY2luZyBkdWkgYmxhbmRpdCB2ZXN0aWJ1bHVtIGJpYmVuZHVtPw=="
+#> [1] "RG9sb3IgbWFsZXN1YWRhIGN1cnN1cyBmYXVjaWJ1cyBmYWNpbGlzaSBhY2N1bXNhbiB2aXZlcnJhPw=="    
+#> [2] "U2l0IHBlbmF0aWJ1cyBsb2JvcnRpcyBhdCBhcHRlbnQgcGVsbGVudGVzcXVlIQ=="                    
+#> [3] "U2l0IGV1aXNtb2QgYWNjdW1zYW4gc2VtcGVyIGFudGUgY3ViaWxpYSBuYW0gdmVsaXQgaGltZW5hZW9zIQ=="
+#> [4] "TG9yZW0gcHVsdmluYXIgYXVndWUgYWxpcXVhbSE="                                            
+#> [5] "RG9sb3IgbnVsbGFtIGZhY2lsaXNpIHNlbmVjdHVzIHBlbmF0aWJ1cy4="
 ```
 
 We can decode all of these using `decode()` as well. This will always
 return a `blob` object.
```
Comment (JosiahParry) [inline README.md]: Can we add a line here something to the effect of 

> "To achieve 0 dependencies, blob is listed only as a suggested dependency. However, the blob is always provided. If you attach the blob package, its print method will be used." 

and then `library(blob)` above the `decode(encoded)`?

File: DESCRIPTION [hunk 3]
```diff
@@ -1,6 +1,6 @@
 Package: b64
 Title: Fast and Vectorized Base 64 Engine
-Version: 0.1.0
+Version: 0.1.0.9000
 Authors@R: 
     person("Josiah", "Parry", , "josiah.parry@gmail.com", role = c("aut", "cre"),
            comment = c(ORCID = "0000-0001-9910-865X"))
```
Comment (JosiahParry) [inline DESCRIPTION]: add yourself at a contributor?

**Comment by JosiahParry** [comment]
Once we fix the vectorized decoding of raws I think this can get merged! Thank you!

**Comment by etiennebacher** [comment]
Thanks for your help @JosiahParry, is this what you had in mind?

``` r
library(b64)
x <- encode(c("hello", "world", NA))
x
#> [1] "aGVsbG8=" "d29ybGQ=" NA

y <- lapply(x, charToRaw)
decode(y)
#> <blob[3]>
#> [[1]]
#> [1] 68 65 6c 6c 6f
#> 
#> [[2]]
#> [1] 77 6f 72 6c 64
#> 
#> [[3]]
#> NULL

# for pretty printing
library(blob)
decode(y)
#> <blob[3]>
#> [1] blob[5 B] blob[5 B] <NA>
```

File: src/rust/src/lib.rs [hunk 4]
```diff
@@ -23,7 +23,7 @@ fn encode_vectorized_(what: Either<Strings, List>, engine: Robj) -> Strings {
     let eng: ExternalPtr<GeneralPurpose> = engine.try_into().unwrap();
     match what {
         Either::Left(s) => s
-            .into_iter()
+            .iter()
```
Comment (JosiahParry) [inline src/rust/src/lib.rs]: can we revert these changes? It's resulting in a 5ms / 12 iter/sec speed reduction.

Comment (CGMossa) [reply src/rust/src/lib.rs]: Good you benchmarked it. I am very surprised. Iter should be more efficient. However, the optimizer is king here..

Comment (etiennebacher) [reply src/rust/src/lib.rs]: Done, I thought `clippy` would get the best thing but apparently not

Comment (JosiahParry) [reply src/rust/src/lib.rs]: Wait? Does this mean I have one over clippy for once in my life....?

Comment (etiennebacher) [reply src/rust/src/lib.rs]: Yes this was done in [634ac3e](https://github.com/extendr/b64/pull/2/commits/634ac3e8b251a8bb792360db41da0bd405eb999e), that only includes changes of `cargo clippy --fix`
